### PR TITLE
Fix reuse of infinity-loader

### PR DESCRIPTION
**Pull request by DuBistKomisch**
So after the refactor to remove `Component#isVisible` usage in https://github.com/ember-infinity/ember-infinity/pull/416 (which was in response to my own issue, so I feel partially responsible), the handling of the `infinityModel` being changed seems to have been broken.

Our use case is paginated search results, so we debounce creating a new infinity model when the user types something.

The first bug was that `_loadStatusDidChange` needed to be called initially for each model change, not just the first, or it could just sit visible on the page saying "Loading..." even if the new model only had one page.

The second bug was that the `_initialInfinityModelSetup` observer was cleared when the element was destroyed, which happens when the component hides itself once reaching infinity. I've moved this cleanup to the component level instead of tying it to the element.

### CODE REVIEW

File: addon/components/infinity-loader.js [hunk 1]
```diff
@@ -127,17 +129,17 @@ const InfinityLoaderComponent = Component.extend({
...
-        infinityModel.off('infinityModelLoaded', instance, instance._loadStatusDidChange.bind(instance));
+        infinityModel.off('infinityModelLoaded', this, this._loadStatusDidChange.bind(this));
       });
 
-    instance.removeObserver('infinityModel', instance, instance._initialInfinityModelSetup);
-    instance.removeObserver('hideOnInfinity', instance, instance._loadStatusDidChange);
-    instance.removeObserver('reachedInfinity', instance, instance._loadStatusDidChange);
+    this.removeObserver('infinityModel', this, this._initialInfinityModelSetup);
+    this.removeObserver('hideOnInfinity', this, this._loadStatusDidChange);
+    this.removeObserver('reachedInfinity', this, this._loadStatusDidChange);
```
Comment (snewcomer) [inline addon/components/infinity-loader.js:142]: 👋 Was this change necessary or could we still use the `will-destroy` modifier?

Comment (snewcomer) [reply addon/components/infinity-loader.js:142]: Thinking out loud...

If the component hides, do we really want to cleanup?  What does it mean to be not visible?  After implementing our own isVisible, we are just using CSS but not actually tearing down the component.

Comment (DuBistKomisch) [reply addon/components/infinity-loader.js:142]: :wave: Yes it was necessary for the second bug mentioned in the OP: `will-destroy` is triggered when setting `isVisible` to `false` which removes the <div> its attached to, which isn't the same thing as the component being destroyed since it can be unhidden again later. Conflating them leaves you in a broken state when it was unhidden since these listeners were missing.
